### PR TITLE
MCC-2223 Limits on fog data in mc-account-key types

### PR DESCRIPTION
**Pull request by jcape**
### Motivation

The artisanal `account_key::{AccountKey, PublicAddress, RootIdentity}` types contain both fixed-length crypto primatives and the variable-length fog URL, report ID, and authority pubkey (or signature over it). This PR enforces some basic limit-based sanity checking on the constructors and conversion traits for those types, making them fallible, and updates all callers.

### In this PR
* `mc_account_keys::limits` is a new private module with relevant constants and "checker" methods.
* `mc_account_keys::error::{Result, Error}` is a new crate-result type / public enum with an enumeration of current errors, and is exported from root.
* `mc_account_keys::constants` contains the default subaddress and (crate-only) subaddress domain tag.
* `mc_account_keys::domain_separators` is deleted.
* `RootIdentity::new_with_fog()` is fallible.
* `AccountKey::new_with_fog()` is now fallible.
* `AccountKey::random_with_fog()` takes explicit fog arguments, and is fallible (this method actually belongs in a `account-key/test-utils` crate)
* `From<&RootIdentity> for AccountKey` is now `TryFrom<&RootIdentity> for AccountKey`
* `PublicAddress::new_with_fog()` is now fallible.
* Fixed minor bug in `mc_util_repr_bytes` where it was using just `Result` in macros, instead of `::core::result::Result`.
* Make unit tests which were instantiating these types use actual `subjectPublicKeyInfo` bytes from test vectors (this will make future improvements to these methods easier).

### Future Work
* Additional validation of fog well-formedness: URL formatting, supported `subjectPublicKeyInfo` types, etc.
* An `is_well_formed()` method on the types which contain fog details, to run these sanity checks on things which were manually constructed.
* Refactor all these types to split out the fog details from the crypto primitives, use types to enforce primitive usages, and design with an eye towards things that would live and run inside of an HSM/enclave, and things which wouldn't.

### CODE REVIEW

File: account-keys/src/limits.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+/// The minimum length of a operator authority's subjectPublicKeyInfo, in bytes.
+///
+/// This is the DER encoded length of an Ed25519 subjectPublicKeyInfo, in bytes.
+const FOG_AUTHORITY_SPKI_MIN_LENGTH: usize = 44;
+
+/// The maximum length of a operator authority's subjectPublicKeyInfo, in bytes
+const FOG_AUTHORITY_SPKI_MAX_LENGTH: usize = 2048;
+
+// /// The maximum length of signature over an operator authority, in bytes
```
Comment (eranrund) [inline account-keys/src/limits.rs:29]: Leftover?

**Review by eranrund — APPROVED** [review]

File: account-keys/src/limits.rs (hunk 1 above)
Comment (jcape) [reply account-keys/src/limits.rs:29]: Yep

**Review by garbageslam — APPROVED** [review]
LGTM, thank you!

**Comment by remoun** [comment]
This PR still seems useful; do we want to merge it, or close it?

**Pull request closed without merging**